src/net/SocketUtil.cxx

```diff
@@ -32,8 +34,21 @@ socket_bind_listen(int domain, int type, int protocol,
 	if (!fd.CreateNonBlock(domain, type, protocol))
 		throw MakeSocketError("Failed to create socket");
 
+#ifdef HAVE_UN
+	if (domain == AF_UNIX) {
+		/* allow everybody to connect */
+		if(fchmod(fd.Get(), 0666) != 0)
+			throw MakeSocketError("fchmod() failed");
+	}
+	else {
+		/* SO_REUSEADDR has no effect on unix sockets */
```

Comment:
This is an unrelated change. And it duplicates code.

Reply:
Why is it unrelated? Do you prefer to keep the code in OneServerSocket::Open()?
Which code is duplicated, what do you mean?

Reply:
Wrong question: you need to explain how this new runtime check for `SetReuseAddress()` is related to switching to `fchmod()`.
You copy'n'pasted the `SetReuseAddress()` call. There are two of them now, one in each `#if` branch. Code duplication is bad.

Reply:
Sure, its possible to remove that duplication but then SetReuseAddress() would run even if its a unix socket. Thats not a problem at all but its also not necessary.
So so question is duplicate code or have code to execute "nops".
I thought the first option would be better but I am fine if you prefer the second.

Reply:
Are you fine with moving that code to SocketUtil or do you want it to stay in ServerSocket?

Reply:
Since this patch is about changing `chmod()` to `fchmod()`, it should do exactly that, and not move code around. Don't do two things at the same time - you make patch review harder than necessary. If you want to move code around or do other things (like adding a runtime check to `SetReuseAddress()`), make that a separate commit.